### PR TITLE
LoadMap InGame #1796

### PR DESCRIPTION
**Pull request by urban119**
loadMap InGame
(now its done... sorry for first upload problems :8ball:)

### CODE REVIEW

**Comment by urban119** [comment]
Game.loadMapU("data/world/castle.otbm", 662, 371)
Map Castle.otmb is position (x=1,y=1)
https://youtu.be/gNEJnYDhcT8?t=46s

**Comment by djarek** [comment]
Almost 400 lines of pure delicious copy pasta and the build fails... Really?

**Comment by urban119** [comment]
cuz create new functions (no take to edit orginally)

File: src/iomap.cpp [hunk 1]
```diff
@@ -475,3 +475,431 @@ bool IOMap::loadMap(Map* map, const std::string& identifier)
...
+	PropStream propStream;
+
+	NODE root = f.getChildNode(nullptr, type);
+	if (!f.getProps(root, propStream)) {
+		setLastErrorString("Could not read root property.");
+		return false;
+	}
+
+	const OTBM_root_header* root_header;
+	if (!propStream.readStruct(root_header)) {
```
Comment (esfomeado) [inline src/iomap.cpp]: if (!propStream.read(root_header)) {

File: src/iomap.cpp [hunk 2]
```diff
@@ -475,3 +475,431 @@ bool IOMap::loadMap(Map* map, const std::string& identifier)
...
+		}
+
+		if (type == OTBM_TILE_AREA) {
+			if (!f.getProps(nodeMapData, propStream)) {
+				setLastErrorString("Invalid map node.");
+				return false;
+			}
+
+			const OTBM_Destination_coords* area_coord;
+			if (!propStream.readStruct(area_coord)) {
```
Comment (esfomeado) [inline src/iomap.cpp]: if (!propStream.read(area_coord)) {

**Comment by urban119** [comment]
I find the me wrong way.... I did not check whether TFS has not been updated...

iomap.cpp has repair now

**Comment by ranisalt** [comment]
Just... STOP programming on the github editor.

**Comment by urban119** [comment]
Im just use that for paste code.

**Comment by marksamman** [comment]
You should reuse the loadMap code, not copy it, just add offsets to the existing function and make them default to 0. The same applies to the loadMap Lua function, we don't need a new function. You should also remove the Urban stuff.

**Pull request closed without merging**